src/OrchardCore.Modules/OrchardCore.Queries/Sql/Drivers/SqlQueryDisplayDriver.cs

```diff
@@ -40,18 +40,16 @@ public override IDisplayResult Edit(SqlQuery query, IUpdateModel updater)
                 // Extract query from the query string if we come from the main query editor
                 if (string.IsNullOrEmpty(query.Template))
                 {
-                    updater.TryUpdateModelAsync(model, "", m => m.Query);
-
-                    // The value is empty for new queries, remove any errors
-                    if (string.IsNullOrEmpty(model.Query))
-                    {
-                        updater.ModelState.Clear();
-                    }
+                    var queryStringModel = new QueryStringModel();
+                    updater.TryUpdateModelAsync(queryStringModel, "", m => m.Query);
+                    model.Query = queryStringModel.Query;
                 }
 
             }).Location("Content:5");
         }
 
+        internal class QueryStringModel { public string Query { get; set; } }
```

Comment:
Why is it needed?

Reply:
It's a fake model **with no required property**. So we can use it for the special case when we try to update the `Query` from the query string, this in case we are coming from the main query editor.

This without adding any error on the curent updater and also without clearing any previous "normal" errors. So it still solves the first issue we had without having to clear the model errors. And it also solves the issue i described above.

But we can change the name, put it elsewhere ...